src/cpp/util/byte_buffer.cc

```diff
@@ -95,4 +102,10 @@ ByteBuffer& ByteBuffer::operator=(const ByteBuffer& buf) {
   return *this;
 }
 
+void ByteBuffer::Swap(ByteBuffer* other) {
+  grpc_byte_buffer* tmp = other->buffer_;
+  other->buffer_ = this->buffer_;
```

Comment:
Thanks for the explanation, and always glad to take performance-enhancing changes. I think we should leave out "this->" for consistency with the rest of the codebase. I've sent in PR #7775 to address that issue.